tests/conftest.py

```diff
@@ -11,6 +11,20 @@
 from trailblazer.mip import files as mip_files_api
 import ruamel.yaml
 
+pytest_plugins = [
```

Comment:
I don't manage to run the tests locally, I get: 
`pytest -v
Traceback (most recent call last):
  File "/Users/patrikgrenfeldt/miniconda3/envs/dev/lib/python3.6/site-packages/_pytest/config/__init__.py", line 430, in _importconftest
    return self._conftestpath2mod[conftestpath]
KeyError: local('/Users/patrikgrenfeldt/Documents/src/cg/tests/conftest.py')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/patrikgrenfeldt/miniconda3/envs/dev/lib/python3.6/site-packages/_pytest/config/__init__.py", line 522, in import_plugin
    __import__(importspec)
ModuleNotFoundError: No module named 'tests.apps'`

Reply:
Have you tried `python setup.py test`?

Reply:
I have now
```
python setup.py test
...
ImportError: Error importing plugin "tests.apps.lims.conftest": No module named 'tests.apps'
```

Reply:
Ok, I guess you need to check your local environment then since it apparently works on travis...

Reply:
not pytest itself atleast, I upgraded it to make sure that wasn't the problem

Reply:
Ok, could you try to reproduce what travis is doing? Didi you run `python setup.py develop` or `pip install --editable .`?

Reply:
I tried to go through the steps of travis, this is what I did: 
  
```
605  git clone --depth=50 --branch=uncouple-loqus https://github.com/Clinical-Genomics/cg.git Clinical-Genomics/cg
  606  conda create decouple-locusdb
  607  conda create -n decouple-locusdb python=3.6
  608  ls
  609  cd Clinical-Genomics/cg/
  610  source activate decouple-loqusdb
  611  source activate decouple-locusdb
  612  git checkout -qf fa2d0b94bb7bcc77b92884c4c231582aff887a36
  613  pip install cython
  614  pip install -r requirements.txt -r requirements-dev.txt -e .
  615  coverage run --source "$(basename "$PWD")" setup.py test
```

This is my result: 
```
running test
Searching for pytest-mock
Best match: pytest-mock 1.10.0
Processing pytest_mock-1.10.0-py3.6.egg

Using /Users/patrikgrenfeldt/Documents/Clinical-Genomics/cg/.eggs/pytest_mock-1.10.0-py3.6.egg
running egg_info
writing cg.egg-info/PKG-INFO
writing dependency_links to cg.egg-info/dependency_links.txt
writing entry points to cg.egg-info/entry_points.txt
writing requirements to cg.egg-info/requires.txt
writing top-level names to cg.egg-info/top_level.txt
reading manifest file 'cg.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
warning: no previously-included files matching '__pycache__' found under directory '*'
writing manifest file 'cg.egg-info/SOURCES.txt'
running build_ext
Traceback (most recent call last):
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/_pytest/config/__init__.py", line 430, in _importconftest
    return self._conftestpath2mod[conftestpath]
KeyError: local('/Users/patrikgrenfeldt/Documents/Clinical-Genomics/cg/tests/conftest.py')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/_pytest/config/__init__.py", line 522, in import_plugin
    __import__(importspec)
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/_pytest/assertion/rewrite.py", line 295, in load_module
    six.exec_(co, mod.__dict__)
  File "/Users/patrikgrenfeldt/Documents/Clinical-Genomics/cg/tests/cli/conftest.py", line 7, in <module>
    from cg.cli import base
  File "/Users/patrikgrenfeldt/Documents/Clinical-Genomics/cg/cg/cli/__init__.py", line 2, in <module>
    from .base import base
  File "/Users/patrikgrenfeldt/Documents/Clinical-Genomics/cg/cg/cli/base.py", line 15, in <module>
    from .upload import upload
  File "/Users/patrikgrenfeldt/Documents/Clinical-Genomics/cg/cg/cli/upload.py", line 9, in <module>
    from cg.apps import coverage as coverage_app, gt, hk, loqus, tb, scoutapi, beacon as beacon_app, \
  File "/Users/patrikgrenfeldt/Documents/Clinical-Genomics/cg/cg/apps/beacon.py", line 5, in <module>
    from cgbeacon.utils import Utility
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/cgbeacon/utils/Utility.py", line 5, in <module>
    from cgbeacon.utils.vcf_panel_filter import vcf_intersect
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/cgbeacon/utils/vcf_panel_filter.py", line 9, in <module>
    from pybedtools import BedTool
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/pybedtools/__init__.py", line 12, in <module>
    from . import contrib
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/pybedtools/contrib/__init__.py", line 3, in <module>
    from . import venn_maker
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/pybedtools/contrib/venn_maker.py", line 12, in <module>
    from pybedtools import helpers
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/pybedtools/helpers.py", line 13, in <module>
    import pysam
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/pysam/__init__.py", line 5, in <module>
    from pysam.libchtslib import *
ImportError: dlopen(/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/pysam/libchtslib.cpython-36m-darwin.so, 2): Library not loaded: @rpath/libcurl.4.dylib
  Referenced from: /Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/pysam/libchtslib.cpython-36m-darwin.so
  Reason: Incompatible library version: libchtslib.cpython-36m-darwin.so requires version 9.0.0 or later, but libcurl.4.dylib provides version 7.0.0

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "setup.py", line 74, in <module>
    'Programming Language :: Python :: Implementation :: CPython',
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/setuptools/__init__.py", line 143, in setup
    return distutils.core.setup(**attrs)
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/distutils/dist.py", line 955, in run_commands
    self.run_command(cmd)
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/setuptools/command/test.py", line 228, in run
    self.run_tests()
  File "setup.py", line 47, in run_tests
    sys.exit(pytest.main(self.test_args))
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/_pytest/config/__init__.py", line 58, in main
    config = _prepareconfig(args, plugins)
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/_pytest/config/__init__.py", line 196, in _prepareconfig
    pluginmanager=pluginmanager, args=args
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/pluggy/hooks.py", line 284, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/pluggy/manager.py", line 67, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/pluggy/manager.py", line 61, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/pluggy/callers.py", line 203, in _multicall
    gen.send(outcome)
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/_pytest/helpconfig.py", line 93, in pytest_cmdline_parse
    config = outcome.get_result()
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/pluggy/callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/_pytest/config/__init__.py", line 675, in pytest_cmdline_parse
    self.parse(args)
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/_pytest/config/__init__.py", line 845, in parse
    self._preparse(args, addopts=addopts)
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/_pytest/config/__init__.py", line 809, in _preparse
    early_config=self, args=args, parser=self._parser
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/pluggy/hooks.py", line 284, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/pluggy/manager.py", line 67, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/pluggy/manager.py", line 61, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/pluggy/callers.py", line 208, in _multicall
    return outcome.get_result()
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/pluggy/callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/_pytest/config/__init__.py", line 719, in pytest_load_initial_conftests
    self.pluginmanager._set_initial_conftests(early_config.known_args_namespace)
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/_pytest/config/__init__.py", line 379, in _set_initial_conftests
    self._try_load_conftest(current)
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/_pytest/config/__init__.py", line 387, in _try_load_conftest
    self._getconftestmodules(x)
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/_pytest/config/__init__.py", line 414, in _getconftestmodules
    mod = self._importconftest(conftestpath)
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/_pytest/config/__init__.py", line 464, in _importconftest
    self.consider_conftest(mod)
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/_pytest/config/__init__.py", line 492, in consider_conftest
    self.register(conftestmodule, name=conftestmodule.__file__)
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/_pytest/config/__init__.py", line 313, in register
    self.consider_module(plugin)
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/_pytest/config/__init__.py", line 498, in consider_module
    self._import_plugin_specs(getattr(mod, "pytest_plugins", []))
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/_pytest/config/__init__.py", line 503, in _import_plugin_specs
    self.import_plugin(import_spec)
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/_pytest/config/__init__.py", line 531, in import_plugin
    six.reraise(new_exc_type, new_exc, sys.exc_info()[2])
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/six.py", line 692, in reraise
    raise value.with_traceback(tb)
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/_pytest/config/__init__.py", line 522, in import_plugin
    __import__(importspec)
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/_pytest/assertion/rewrite.py", line 295, in load_module
    six.exec_(co, mod.__dict__)
  File "/Users/patrikgrenfeldt/Documents/Clinical-Genomics/cg/tests/cli/conftest.py", line 7, in <module>
    from cg.cli import base
  File "/Users/patrikgrenfeldt/Documents/Clinical-Genomics/cg/cg/cli/__init__.py", line 2, in <module>
    from .base import base
  File "/Users/patrikgrenfeldt/Documents/Clinical-Genomics/cg/cg/cli/base.py", line 15, in <module>
    from .upload import upload
  File "/Users/patrikgrenfeldt/Documents/Clinical-Genomics/cg/cg/cli/upload.py", line 9, in <module>
    from cg.apps import coverage as coverage_app, gt, hk, loqus, tb, scoutapi, beacon as beacon_app, \
  File "/Users/patrikgrenfeldt/Documents/Clinical-Genomics/cg/cg/apps/beacon.py", line 5, in <module>
    from cgbeacon.utils import Utility
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/cgbeacon/utils/Utility.py", line 5, in <module>
    from cgbeacon.utils.vcf_panel_filter import vcf_intersect
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/cgbeacon/utils/vcf_panel_filter.py", line 9, in <module>
    from pybedtools import BedTool
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/pybedtools/__init__.py", line 12, in <module>
    from . import contrib
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/pybedtools/contrib/__init__.py", line 3, in <module>
    from . import venn_maker
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/pybedtools/contrib/venn_maker.py", line 12, in <module>
    from pybedtools import helpers
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/pybedtools/helpers.py", line 13, in <module>
    import pysam
  File "/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/pysam/__init__.py", line 5, in <module>
    from pysam.libchtslib import *
ImportError: Error importing plugin "tests.cli.conftest": dlopen(/Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/pysam/libchtslib.cpython-36m-darwin.so, 2): Library not loaded: @rpath/libcurl.4.dylib
  Referenced from: /Users/patrikgrenfeldt/miniconda3/envs/decouple-locusdb/lib/python3.6/site-packages/pysam/libchtslib.cpython-36m-darwin.so
  Reason: Incompatible library version: libchtslib.cpython-36m-darwin.so requires version 9.0.0 or later, but libcurl.4.dylib provides version 7.0.0
```

Reply:
Ok so there seems to be a dependency conflict in cg not related to the tests? I'm not sure how to handle this or why neither me or travis get's the same error...

Reply:
@ingkebil, can you run the tests locally?

Reply:
Yes, it works for me with the instructions in this thread.

However, additionally, I had to install:
- `pip install numpy`
- `conda isntall ncurses`

for this to work.

Reply:
I'm not sure why I had to install numpy explicitly. It seems the cyvcf2 requires numpy and somehow numpy only appeared to be installed _after_ cyvcf2.

Reply:
I suggest that master should still be runable out of the box on any of our machines, our what do you say?

Reply:
How moonso helped me to get the tests up and running again:

1.  Update curl
2. Rebuild pysam with the new version of curl (pip uninstall then pip install with nocache option)